File: Planets/Thear/Biome.js
const fs = require('fs');

class Biome {
    constructor(name) {
        this.name = name;
        this.organisms = [];
    }

    loadOrganismsFromJson(filePath) {
        return JSON.parse(fs.readFileSync(filePath, 'utf8'));
    }

    addOrganismsFromJson(filePath) {
        const organismsData = this.loadOrganismsFromJson(filePath);
        organismsData.forEach(organismData => {
            let organism;
            if (organismData.hasOwnProperty('limbs')) {
                organism = new Animal(organismData.species, organismData.age, organismData.limbs, organismData.habitat);
            } else {
                organism = new Plant(organismData.species, organismData.age, organismData.habitat);
            }
            this.addOrganism(organism);
        });
    }

    addOrganism(organism) {
        this.organisms.push(organism);
    }

    simulateDay() {
        console.log(`Simulating a day in the ${this.name} Biome:`);
        this.organisms.forEach(organism => {
            if (organism.isAlive) {
                organism.move();
                organism.eat();
                organism.reproduce();
                if (organism instanceof Animal) {
                    const prey = this.findPrey(organism);
                    if (prey) {
                        console.log(`${organism.species} is hunting ${prey.species}.`);
                        prey.die();
                        // Implement limb removal for the prey
                        prey.loseLimb();
                    }
                    // Simulate animal communication
                    this.animalCommunication(organism);
                }
            } else {
                console.log(`${organism.species} is dead and cannot perform any actions.`);
            }
        });
    }

    findPrey(predator) {
        const preyCandidates = this.organisms.filter(org => org instanceof Animal && org !== predator);
        if (preyCandidates.length > 0) {
            return preyCandidates[Math.floor(Math.random() * preyCandidates.length)];
        }
        return null;
    }

    animalCommunication(organism) {
        // Simulate animal communication
        console.log(`${organism.species} is communicating with other ${organism.species}.`);
    }
}

class Organism {
    constructor(species, age) {
        this.species = species;
        this.age = age;
        this.isAlive = true;
    }

    move() {
        console.log(`${this.species} is moving.`);
    }

    eat() {
        console.log(`${this.species} is eating.`);
    }

    reproduce() {
        console.log(`${this.species} is reproducing.`);
    }

    die() {
        console.log(`${this.species} has died.`);
        this.isAlive = false;
    }
}

class Plant extends Organism {
    constructor(species, age, habitat) {
        super(species, age);
        this.habitat = habitat;
    }

    move() {
        console.log(`${this.species} can't move, it's a plant in the ${this.habitat} habitat.`);
    }

    eat() {
        console.log(`${this.species} is absorbing nutrients from the soil in the ${this.habitat} habitat.`);
    }

    reproduce() {
        console.log(`${this.species} is spreading seeds in the ${this.habitat} habitat.`);
    }
}

class Animal extends Organism {
    constructor(species, age, limbs, habitat) {
        super(species, age);
        this.limbs = limbs;
        this.habitat = habitat;
    }

    move() {
        console.log(`${this.species} is moving in the ${this.habitat} habitat.`);
    }

    eat() {
        console.log(`${this.species} is hunting for food.`);
    }

    reproduce() {
        console.log(`${this.species} is mating.`);
    }

    loseLimb() {
        if (this.limbs > 0) {
            this.limbs--;
            console.log(`${this.species} lost a limb! Remaining limbs: ${this.limbs}`);
        } else {
            this.die();
        }
    }
}

// Sample usage
const biomeForest = new Biome("Forest");
biomeForest.addOrganismsFromJson('animals.json');
biomeForest.addOrganismsFromJson('plants.json');

biomeForest.simulateDay();

File: Planets/Thear/Biome.py
# File = robjam1990/Psychosis/Gameplay/Planets/Thear/Biome.py
import random
import json

class Biome:
    def __init__(self, name):
        self.name = name
        self.organisms = []

    def load_organisms_from_json(self, file_path):
        with open(file_path, 'r') as file:
            data = json.load(file)
        return data

    def add_organisms_from_json(self, file_path):
        organisms_data = self.load_organisms_from_json(file_path)
        for organism_data in organisms_data:
            if 'limbs' in organism_data:  # Assuming 'limbs' only for animals
                organism = Animal(organism_data['species'], organism_data['age'], organism_data['limbs'], organism_data['habitat'])
            else:
                organism = Plant(organism_data['species'], organism_data['age'], organism_data['habitat'])
            self.add_organism(organism)

    def add_organism(self, organism):
        self.organisms.append(organism)

    def simulate_day(self):
        print(f"Simulating a day in the {self.name} Biome:")
        for organism in self.organisms:
            if organism.is_alive:
                organism.move()
                organism.eat()
                organism.reproduce()
                if isinstance(organism, Animal):
                    prey = self.find_prey(organism)
                    if prey:
                        print(f"{organism.species} is hunting {prey.species}.")
                        prey.die()
                        # Implement limb removal for the prey
                        prey.lose_limb()
                    # Simulate animal communication
                    self.animal_communication(organism)
            else:
                print(f"{organism.species} is dead and cannot perform any actions.")

    def find_prey(self, predator):
        prey_candidates = [org for org in self.organisms if isinstance(org, Animal) and org != predator]
        if prey_candidates:
            return random.choice(prey_candidates)
        return None

    def animal_communication(self, organism):
        # Simulate animal communication
        print(f"{organism.species} is communicating with other {organism.species}.")

class Organism:
    def __init__(self, species, age):
        self.species = species
        self.age = age
        self.is_alive = True

    def move(self):
        print(f"{self.species} is moving.")

    def eat(self):
        print(f"{self.species} is eating.")

    def reproduce(self):
        print(f"{self.species} is reproducing.")

    def die(self):
        print(f"{self.species} has died.")
        self.is_alive = False

class Plant(Organism):
    def __init__(self, species, age, habitat):
        super().__init__(species, age)
        self.habitat = habitat

    def move(self):
        print(f"{self.species} can't move, it's a plant in the {self.habitat} habitat.")

    def eat(self):
        print(f"{self.species} is absorbing nutrients from the soil in the {self.habitat} habitat.")

    def reproduce(self):
        print(f"{self.species} is spreading seeds in the {self.habitat} habitat.")

class Animal(Organism):
    def __init__(self, species, age, limbs, habitat):
        super().__init__(species, age)
        self.limbs = limbs
        self.habitat = habitat

    def move(self):
        print(f"{self.species} is moving in the {self.habitat} habitat.")

    def eat(self):
        print(f"{self.species} is hunting for food.")

    def reproduce(self):
        print(f"{self.species} is mating.")

    def lose_limb(self):
        if self.limbs > 0:
            self.limbs -= 1
            print(f"{self.species} lost a limb! Remaining limbs: {self.limbs}")
        else:
            self.die()

# Sample usage
biome_forest = Biome("Forest")
biome_forest.add_organisms_from_json('animals.json')
biome_forest.add_organisms_from_json('plants.json')

biome_forest.simulate_day()
